mautic/templates/deployment.yaml

```diff
@@ -34,7 +42,10 @@ spec:
             - name: MAUTIC_DB_USER 
               value: {{ .Values.database.user | quote }}
             - name: MAUTIC_DB_PASSWORD
-              value: {{ .Values.database.password | quote }}
+              valueFrom:
```

Comment:
We need to keep old logic, where we can provide the password with the value.
If it is empty, lets use secret.